code/processes/wdb.q

```diff
@@ -258,28 +258,35 @@ movetohdb:{[dw;hw;pt]
      .lg.e[`mvtohdb;raze"Table(s) ",string[(key hsym`$hw)inter(key hsym`$dw)]," is present in both location. Operation will be aborted to avoid corrupting the hdb"]]
  }
 
+reloadsymfile:{[symFilePath]
+  .lg.o[`sort; "reloading the sym file from: ",string symFilePath];
+  @[load; symFilePath; {.lg.e[`sort;"failed to reload sym file: ",x]}]
+ }
+
 endofdaysortdate:{[dir;pt;tablist;hdbsettings]
-	/-sort permitted tables in database
-	/- sort the table and garbage collect (if enabled)
-	.lg.o[`sort;"starting to sort data"];
-	$[count[.z.pd[]]&0>system"s";
-		[.lg.o[`sort;"sorting on slave sort", string .z.p];
-		{[x;compression] setcompression compression;.sort.sorttab x;if[gc;.gc.run[]]}[;hdbsettings`compression] peach tablist,'.Q.par[dir;pt;] each tablist];
-		[.lg.o[`sort;"sorting on master sort"];
-		{[x] .sort.sorttab[x];if[gc;.gc.run[]]} each tablist,'.Q.par[dir;pt;] each tablist]];
-	.lg.o[`sort;"finished sorting data"];
+  /-sort permitted tables in database
+  /- sort the table and garbage collect (if enabled)
+  .lg.o[`sort;"starting to sort data"];
+  $[count[.z.pd[]]&0>system"s";
+    [.lg.o[`sort;"sorting on slave sort", string .z.p];
+     {(neg x)(`.wdb.reloadsymfile;.Q.dd[.wdb.hdbdir;`sym])} each .z.pd[];
```

Comment:
The sym file path should be passed in to the remote call rather than using the local .wdb.hdbdir variable on the remote process, as these might be different. Also, use hdbsettings`hdbdir. So it should be

```{(neg x)(`.wdb.reloadsymfile;y)}[;.Q.dd[hdbsettings`hdbdir;`sym]] each .z.pd[]```